### DISCUSSION TITLE
When moving into a room - does "look" command get called or does "return_appearance" get returned?

### DISCUSSION THREAD

**Discussion by RodRitter** [evennia/evennia#2659] [Evennia Questions] [answered]
Hey all,

I'm having trouble implementing a prompt that appears after every command. I've overwritten all commands, including the "look" command to include this prompt using 'at_post_cmd'.

The problem is that the prompt isn't displaying after moving. It does if I type "look" - I assumed look command would be executed after a move into a room, but looks like that's not the case.

**Comment by jaborsh** [comment, marked as answer]
Hey! Good question.

The following can be found in the `DefaultCharacter` typeclass:

    `def at_post_move(self, source_location, **kwargs):`
        """
        We make sure to look around after a move.

        """
        `if self.location.access(self, "view"):
            self.msg(text=(self.at_look(self.location), {"type": "look"}))`

**Reply by jaborsh** [reply]
Take note that this is from the Develop branch. 0.9.5 will have the same functionality but the hook is named something else. If memory serves, should be `at_after_move`.

**Reply by RodRitter** [reply]
I know this is not an issues thread so I hope I'm not too out of line here:|

I've implemented the following:

File: `command.py`
```
class Command(default_cmds.MuxCommand):
    # Vitals prompt after every command
    def at_post_cmd(self):
        hp = f"|=mHP:|411{self.caller.db.hp}/{self.caller.db.max_hp}"
        mp = f"|=mMP:|225{self.caller.db.mp}/{self.caller.db.max_mp}"
        st = f"|=mST:|430{self.caller.db.st}/{self.caller.db.max_st}"
        self.caller.msg(prompt=f"{hp}  {mp}  {st}\n")

class CmdLook(default_cmds.CmdLook, Command):
    pass
```

If I execute `look`, it shows correctly:
```
... # Room appearance stuff here
HP:100/100  MP:100/100  ST:100/100
```

If I move - `east`, I don't see the prompt:
```
... # Room appearance stuff here
# no prompt here
```

**Reply by jaborsh** [reply]
Oh. I see what the issue is.

When you are moving, you are not inherently executing the `look` command. Rather, you are using the `at_look` hook.

Either add your prompt to `at_post_move` or edit the `at_look` hook to display your desired messages.

**Reply by Griatch** [reply]
Here's the docs on how exits work: https://www.evennia.com/docs/latest/Objects.html?highlight=at_after_move#exits 

As seen, `look` is not invoked. As @The-Archemist mentions, you need to override the `at_after_move` hook in this case. 

Overall, the tutorial for making a prompt suggests you instead override `msg` if you want to have a prompt appear after every command input. https://www.evennia.com/docs/latest/Command-Prompt.html

**Reply by RodRitter** [reply]
Thank you both for your replies. They're very helpful
